app.rb

```diff
@@ -0,0 +1,160 @@
+require_relative 'classes/student'
+require_relative 'classes/teacher'
+require_relative 'classes/book'
+require_relative 'classes/rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'All books:'
+    @books.each do |book|
+      puts "ID: #{book.id}.- #{book.title} by #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'All people:'
+    @people.each do |person|
+      puts "ID: #{person.id}.- #{person.name} (#{person.class})"
+    end
+  end
+
+  def create_person
+    puts 'Enter person name:'
+    name = gets.chomp
+    puts 'Enter person age:'
+    age = gets.chomp.to_i
+    puts 'Enter person role:'
+
+    puts '1. Student'
+    puts '2. Teacher'
+    choice = gets.chomp.to_i
+
+    person_of_type(choice, name, age)
+  end
+
+  def person_of_type(choice, name, age)
+    case choice
+    when 1
+      @people.push(Student.new(name, age))
+      puts "New student '#{name}' created successfully!"
+    when 2
+      puts 'Enter teacher specialization'
+      specialization = gets.chomp
+      @people.push(Teacher.new(name, age, specialization))
+      puts "New teacher '#{name}' created successfully!"
+    else
+      puts 'Invalid choice. Person was not created'
+    end
```

Comment:
Great work done by handling the errors 👍🏼  👍🏼

Reply:
Thank you very much!